### ISSUE TITLE
Always in bold face when the text is Chinese

### ISSUE THREAD

**Issue by icoigo** [AliSoftware/OHAttributedLabel#33]
No matter what CTFontUIFontType I set to the attributed string, as long as the string is Chinese, the text will be showed in bold. I don't know if other language has the same problem.

**Comment by AliSoftware** [comment]
No problem with any other language in my code, even in korean (but never tried chinese specifically)
- With which CTFonts (or UIFont) did you try?
- And which method do you try to create your font? `CTFontCreateUIFontForLanguage`? Or you used a CTFont with an explicit name? or a FontDescriptor? Or do you use an UIFont and used e.g. my commodity category "NSAttributedString+Attributes" to set the font using `-(void)setFont:(UIFont*)font;`? Please give details.

Anyway, as I don't do anything special for Chinese differenty and I manage lots of other Unicode texts without a problem, I bet the problem is related to CoreText itself.

Once you create your CTFont, check its `CTFontDescriptorRef`, and especially in the kCTFontTraitsAttribute of the FontDescription, this description probably tells you that the font trait is bold, so that would be normal.
Apple probably has only one CTFont that exactly match Chinese when you use `CTFontCreateUIFontForLanguage` so that's why this method always return the same font, which you find to be bold…but many other fonts (if not all) support Unicode, so there is no reason not to use any other fonts in your case!

**Comment by icoigo** [comment]
Great detail and quick response. I will check that out and get back to post more information as soon as I can.
Thank you very much.

**Comment by icoigo** [comment]
Hi,
I use CTFontCreateUIFontForLanguage to create the CTFont. I try to use two font as follow:
1. CTFontCreateUIFontForLanguage(kCTFontEmphasizedSystemFontType, 13.0, NULL);
2. CTFontCreateUIFontForLanguage(kCTFontSystemFontType, 13.0, NULL);

The first one's font traits is showed as follow:
 NSCTFontProportionTrait = 0;
 NSCTFontSlantTrait = 0;
 NSCTFontSymbolicTrait = 2;
 NSCTFontWeightTrait = "0.4";

second one's font traits is showed as follow:
NSCTFontProportionTrait = 0;
NSCTFontSlantTrait = 0;
NSCTFontSymbolicTrait = 0;
NSCTFontWeightTrait = 0;

So, the font returned is actually different. I notice that, if the string has English and Chinese mixed, like this "English 中文", the English part will be showed different, which is bold and not bold, but the Chinese part is the same, no difference at all.

Is this the core text problem?

**Comment by AliSoftware** [comment]
OK. Yes it seems to be a bug in CoreText.

You should probably be able to use a CATextLayer in a test app and set its text attribute to your NSAttributedString, then display this CATextLayer on screen.
This way, even if it won't perform framing, wordwrapping and such things OHAL does, you should at least be able to display a simple chinese NSAttributedString on screen and check if you still have the problem. If so, this would confirm that this is a CoreText problem.

**Comment by icoigo** [comment]
I have tried to use CATextLayer, and the text (although pure English) looks blurry.

**Comment by AliSoftware** [comment]
Seems to be an antialiasing problem on your CALayer rendering. Don't have time to help right now sorry.

Anyway, do simply try with another font, without using `CTFontCreateUIFontForLanguage`.
Create the UIFont you like e.g. using `systemFontWithSize:` or whatever method of UIFont, then user `setFont:` on the `NSMutableAttributedString`

**Comment by icoigo** [comment]
I have searched around and finally I solved the blurry text problem with CATextLayer, just need to set the correct contentsScale to make the text be clear. However, CATextLayer has the same problem as using the Core Text drawing. English text is cool but not for the Chinese, it shows in bold.
I have also tried to use your category method, setFont:(UIFont *), but the result is the same. When I looked into your source codes, I have tried this way before actually.
So this could be the core text problem, or there is a trick we do not know can fix this.
